articles/10-tips-for-working-with-node.js/index.md

```diff
@@ -0,0 +1,295 @@
+```js
+*// writing a hello world to console: *
+console.log(\"hello world\");
+```
+`Console.log` is provided by internet browsers such as Chrome DevTools, it has debug protocols ported into a Node.js module and can be used to debug Node apps. 
+
+It requires you to `install node -inspector` (`npm install -g node-inspector`) then run ` node-debug app.js --debug-brk`. 
+This will open up Chrome Developer Tools and you can start debugging your app. 
+
+Alternatively, you can use Node.js' built-in debugger by running `node debug app.js`.
+
+Another method is using `watchers` in `expressions` and `variables`. To start
+watching an `expression`, include `watch ('my expression');`. 
+
+The command `watcher` will print active watchers. To remove a watcher you can use: `unwatch ('my expression');` Check out this article to find [more efficient ways to deal with bugs from your app](https://www.section.io/engineering-education/debug-nodejs-vscode/).
+
+## 7. Always Keep Your Code Simple and Light
+
+It is important to keep the Node.js codebase compact and shallow to reduce the latency and speed up the response time of the application. Some devices are slower and their latencies are higher thus you need to keep the code small, well-arranged executions and light. 
+
+These same ideas should be applied to your server code as well. While writing your Node.js code ensure proper and appropriate naming of things for better referencing.
+
+This will serve as documentation for your application and make it easier to go back to your code when changes are required. Better code quality helps in easier code maintainability
+
+## 8. Security practices for your server
+
+As you get started with Node.js, try to secure your application by using proper coding, tooling, and operations.
+
+The main practices that will help to develop secure Node.js application include:
+
+-   Set up secure HTTP headers.
+
+-   Do a static review of your codebase.
+
+-   Handle errors carefully (error codes and stack traces).
+
+-   Regularly use strict mode i.e. restricted variant, undeletable
+    properties, validation, and authentication.
+
+-   Implement proper session management such as HTTP only.
+
+-   Set cookie scope and cookie flags.
+
+-   Avoid command injection.
+
+-   Check for any vulnerabilities using `retire.js`. Retire.js will assist
+    you to recognize module versions with associated vulnerabilities. Install it using `npm install -g retire`. After running
+    it, the `retire` command will scan for any vulnerabilities present in your application.
+
+-   Constantly audit all the modules you have in your Node.js application using CLI Node Security Platform. Node Security CLI uses the command `nsp`, which allows auditing of `package.json` and `npm-shrinkwrap.json` to check unsafe modules. Run `npm install nsp -global` from your application project directory and `nsp check` to start the check.
```

Comment:
This is depreciated and will be removed soon. This functionality is built into npm using `npm audit`. See this [blog post](https://blog.npmjs.org/post/175511531085/the-node-security-platform-service-is-shutting).